ksql-common/src/main/java/io/confluent/ksql/schema/ksql/LogicalSchema.java

```diff
@@ -319,9 +335,22 @@ public static boolean isImplicitColumnName(final String fieldName) {
     return IMPLICIT_FIELD_NAMES.contains(fieldName.toUpperCase());
   }
 
-  @SuppressWarnings("UnstableApiUsage")
+  private Set<String> implicitFieldNames() {
+    return fieldNames(implicitFields());
+  }
+
+  private Set<String> keyFieldNames() {
+    return fieldNames(keyFields());
+  }
+
   private Set<String> implicitAndKeyFieldNames() {
-    return Streams.concat(implicitFields().stream(), keyFields().stream())
+    final Set<String> names = implicitFieldNames();
+    names.addAll(keyFieldNames());
+    return names;
+  }
+
+  private static Set<String> fieldNames(final Collection<Field> fields) {
+    return fields.stream()
```

Comment:
since this class is immutable, I feel like we should compute these all at construction instead of every time we call the methods

Reply:
We could. I was thinking the same.  This class will hopefully soon be re-written with its own Field and Schema type. Can address it then.